### ISSUE TITLE
Easy way to calculate the gazelle name of something given the import path

### ISSUE THREAD

**Issue by jeb2239** [bazelbuild/bazel-gazelle#202]
Is this at all possible? without having to set up a whole workspace and everything.

**Comment by jayconrod** [comment]
I'm assuming you mean how to resolve an import path to a Bazel label. Please clarify if not.

The rules for this have gotten a bit complex, but conventions are usually followed, so it's pretty predictable.

* You need a *prefix* (set with `-go_prefix` on the command line or `# gazelle:prefix` in a build file or `importpath` in `go_repository`) and the import path.
* The first part of the label is the package name where the prefix is set. For `-go_prefix`, this is `//`. For `go_repository`, this is `@name_of_repo//`.
* If you don't know the name of the repository (it isn't declared in WORKSPACE), Gazelle will go and find the repository root (using `golang.org/x/tools/go/vcs`), then convert that to something that looks like a repo name (e.g., `github.com/pkg/errors` => `com_github_pkg_errors`)
* The rest of the label is everything in the `importpath` after the prefix, plus `:go_default_library`.

For development repositories, Gazelle will scan library rules, build a table mapping `importpath` strings to labels, and will use that. So the mapping can be completely arbitrary.

For more info see the [design doc](Design.rst). For details, see the `internal/resolve` package.

**Comment by jeb2239** [comment]
yeah I guess I want an accurate way to get the exact tag gazelle will calculate from a given import path.

**Comment by jeb2239** [comment]
i want something like wtool -asis , but prints to stdout

**Comment by jeb2239** [comment]
yeah but I am trying to get the label gazelle will use the com_github_whatever thing

**Comment by jeb2239** [comment]
but for really complex scenarios it is hard to guess what gazzelle will do
